src/plugins/appodeal.ts

```diff
@@ -0,0 +1,443 @@
+   * @param {boolean} enabled
+   */
+  @Cordova()
+  static enableSkippableVideoCallbacks(enabled: boolean): void {};
+
+  /**
+   * enable event listeners
+   * @param {boolean} enabled
+   */
+  @Cordova()
+  static enableNonSkippableVideoCallbacks(enabled: boolean): void {};
+
+  /**
+   * enable event listeners
+   * @param {boolean} enabled
+   */
+  @Cordova()
+  static enableBannerCallbacks(enabled: boolean): void {};
+
+  /**
+   * enable event listeners
+   * @param {boolean} enabled
+   */
+  @Cordova()
+  static enableRewardedVideoCallbacks(enabled: boolean): void {};
+
+  /**
+   *
+   * @param {string} name - name of rule
+   * @param {boolean} value
+   */
+  @Cordova()
+  static setCustomBooleanRule(name: string, value: boolean): void {};
+
+  /**
+   *
+   * @param {string} name - name of rule
+   * @param {number} value
+   */
+  @Cordova()
+  static setCustomIntegerRule(name: string, value: number): void {};
+
+  /**
+   * set rule with float value
+   * @param {string} name
+   * @param {number} value
+   */
+  @Cordova()
+  static setCustomDoubleRule(name: string, value: number): void {};
+
+  /**
+   * set rule with string value
+   * @param {string} name - name of rule
+   * @param {string} value
+   */
+  @Cordova()
+  static setCustomStringRule(name: string, value: string): void {};
+
+  /**
+   * set ID preference in Appodeal for current user
+   * @param id
+   */
+  @Cordova()
+  static setUserId(id: any): void {};
+
+  /**
+   * set Email preference in Appodeal for current user
+   * @param email
+   */
+  @Cordova()
+  static setEmail(email: any): void {};
+
+  /**
+   * set Birthday preference in Appodeal for current user
+   * @param birthday
+   */
+  @Cordova()
+  static setBirthday(birthday: any): void {};
+
+  /**
+   * et Age preference in Appodeal for current user
+   * @param age
+   */
+  @Cordova()
+  static setAge(age: any): void {};
+
+  /**
+   * set Gender preference in Appodeal for current user
+   * @param gender
+   */
+  @Cordova()
+  static setGender(gender: any): void {};
+
+  /**
+   * set Occupation preference in Appodeal for current user
+   * @param occupation
+   */
+  @Cordova()
+  static setOccupation(occupation: any): void {};
+
+  /**
+   * set Relation preference in Appodeal for current user
+   * @param relation
+   */
+  @Cordova()
+  static setRelation(relation: any): void {};
+
+  /**
+   * set Smoking preference in Appodeal for current user
+   * @param smoking
+   */
+  @Cordova()
+  static setSmoking(smoking: any): void {};
+
+  /**
+   * set Alcohol preference in Appodeal for current user
+   * @param alcohol
+   */
+  @Cordova()
+  static setAlcohol(alcohol: any): void {};
+
+  /**
+   * set Interests preference in Appodeal for current user
+   * @param interests
+   */
+  @Cordova()
+  static setInterests(interests: any): void {};
+
+  /*********************
+   * event Observables *
+   *********************/
+
+  static onInterstitialLoaded(): Observable<any> {
```

Comment:
instead of manually writing the function, you can add the following decorator:
```typescript
@Cordova({
  eventObservable: true,
  event: 'onInterstitialLoaded'
})
```